atoma-proxy/src/server/middleware.rs

```diff
@@ -587,20 +613,103 @@ pub async fn handle_locked_stack_middleware(
         })?;
     let original_req = Request::from_parts(req_parts.clone(), Body::from(body_bytes.clone()));
     let response = next.clone().run(original_req).await;
-    if response.status() == StatusCode::LOCKED {
-        // Lock the current stack, in the Proxy's internal state
-        utils::lock_stack(&state.state_manager_sender, &mut req_parts, &endpoint)?;
-        req_parts
-            .headers
-            .insert(AUTHORIZATION, authorization_header);
-        let req = Request::from_parts(req_parts, Body::from(body_bytes));
-        if is_confidential_compute_endpoint(&endpoint) {
-            confidential_compute_middleware(state, req, next).await
-        } else {
+    match response.status() {
+        StatusCode::LOCKED => {
+            // Lock the current stack, in the Proxy's internal state
+            utils::lock_stack(&state.state_manager_sender, &mut req_parts, &endpoint)?;
+            req_parts
+                .headers
+                .insert(AUTHORIZATION, authorization_header);
+            let req = Request::from_parts(req_parts, Body::from(body_bytes));
+            if is_confidential_compute_endpoint(&endpoint) {
+                // NOTE: If this is a confidential compute request, the client already sent a request encrypted for a specific node.
+                // Acquiring a new stack in this case might simply select a new node, which means that the client will have to re-encrypt the request,
+                // with the new node's public key. For this reason, instead of being the proxy retrying to acquire a new stack, we let the client retry
+                // a couple of times, through Atoma's native SDKs.
+                return Err(AtomaProxyError::Locked {
+                    message: "Confidential compute requests should be retried".to_string(),
```

Comment:
Let's change the error code here to be more specific to the `LOCKED` scenario.

Reply:
You mean the message ?

Reply:
Sorry yes I meant the message, both the `LOCKED` and `TOO_EARLY` say `Confidential compute requests should be retried`